src/main.rs

```diff
@@ -70,24 +70,16 @@ fn main() -> Result<()> {
     // Setup app
     let mut app = App::new(env.clone(), &mut commander)?;
 
-    // Setup terminal
-    enable_raw_mode()?;
-    let mut stdout = io::stdout();
-    execute!(stdout, EnterAlternateScreen, EnableMouseCapture)?;
-    let backend = CrosstermBackend::new(stdout);
-    let mut terminal = Terminal::new(backend)?;
+    let mut terminal = setup_terminal()?;
 
     // Run app
-    run_app(&mut terminal, &mut app, &mut commander)?;
-
-    // Restore terminal
-    disable_raw_mode()?;
-    execute!(
-        terminal.backend_mut(),
-        LeaveAlternateScreen,
-        DisableMouseCapture
-    )?;
-    terminal.show_cursor()?;
+    match run_app(&mut terminal, &mut app, &mut commander) {
+        Ok(()) => restore_terminal(terminal)?,
+        e @ Err(_) => {
+            restore_terminal(terminal)?;
+            return e;
+        },
+    }
```

Comment:
That's also possible to do something like


```suggestion
    let res = run_app(&mut terminal, &mut app, &mut commander);
    restore_terminal(terminal)?;
    res?;
```

What do you think?

Reply:
Yes, I think that's simpler. I've never seen the `e @ Err(_)` syntax before either.